### PR TITLE
fix(auth): handle tooManyRequests and limitExceeded separately in AuthErrorHelper

### PR DESCRIPTION
**Pull request by thisisabhash**
*Issue #, if available:* 1041

*Description of changes:* Removed the coupling tooManyRequests and limitExceeded in the switch block inside AuthErrorHelper 

*Check points:*

~~Added new tests to cover change, if needed~~
- [x] All unit tests pass
- [x] All integration tests pass
~~Documentation update for the change if required~~
- [x] PR title conforms to conventional commit style

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by royjit** [comment]
The unit test failed because of the change. Also, please mark the check points and strike out whichever are not required.

**Comment by thisisabhash** [comment]
Marked the checkpoints. Fixed the UT as well.

File: AmplifyPlugins/Auth/AWSCognitoAuthPlugin/Support/Constants/AuthPluginErrorConstants.swift [hunk 1]
```diff
@@ -229,7 +229,11 @@ extension AuthPluginErrorConstants {
     static let tooManyFailedError: RecoverySuggestion = "User might have tried too many times with failed input"
 
     static let tooManyRequestError: RecoverySuggestion = """
-    Make sure the requests send are controlled and the errors are properlly handled
+    Make sure the requests send are controlled and the errors are properly handled
+    """
+
+    static let limitExceededError: RecoverySuggestion = """
+    User might have exceeded the resource limit of the requested AWS resource
```
Comment (royjit) [inline AmplifyPlugins/Auth/AWSCognitoAuthPlugin/Support/Constants/AuthPluginErrorConstants.swift:236]: Word it as a recovery suggestion - "Make sure that the request made to the particular AWS resources are under the resource quota limits"

File: AmplifyPlugins/Auth/AWSCognitoAuthPluginTests/AuthUserBehaviorTests/UserBehaviorChangePasswordTests.swift [hunk 2]
```diff
@@ -149,7 +149,7 @@ class UserBehaviorChangePasswordTests: BaseUserBehaviorTest {
     /// - When:
     ///    - I invoke changePassword with old password and new password
     /// - Then:
-    ///    - I should get a .requestLimitExceeded error
+    ///    - I should get a .limitExceeded error
     ///
     func testChangePasswordWithLimitExceededException() {
```
Comment (royjit) [inline AmplifyPlugins/Auth/AWSCognitoAuthPluginTests/AuthUserBehaviorTests/UserBehaviorChangePasswordTests.swift:154]: Add test for `requestLimitExceeded` exception also.

File: AmplifyPlugins/Auth/AWSCognitoAuthPluginTests/AuthUserBehaviorTests/UserBehaviorChangePasswordTests.swift [hunk 3]
```diff
@@ -168,7 +168,7 @@ class UserBehaviorChangePasswordTests: BaseUserBehaviorTest {
                     XCTFail("Should produce service error instead of \(error)")
                     return
                 }
-                guard case .requestLimitExceeded = (underlyingError as? AWSCognitoAuthError) else {
+                guard case .limitExceeded = (underlyingError as? AWSCognitoAuthError) else {
                     XCTFail("Underlying error should be requestLimitExceeded \(error)")
```
Comment (royjit) [inline AmplifyPlugins/Auth/AWSCognitoAuthPluginTests/AuthUserBehaviorTests/UserBehaviorChangePasswordTests.swift:172]: Wording "Underlying error should be limitExceeded \(error)"

File: AmplifyPlugins/Auth/AWSCognitoAuthPlugin/Models/Error/AWSCognitoAuthError.swift [hunk 4]
```diff
@@ -34,7 +34,7 @@ public enum AWSCognitoAuthError: Error {
     /// Password given is invalid.
     case invalidPassword
 
-    /// Number of allowed operation have exceeded.
+    /// The user has exceeded the limit for a requested AWS resource.
```
Comment (royjit) [inline AmplifyPlugins/Auth/AWSCognitoAuthPlugin/Models/Error/AWSCognitoAuthError.swift:37]: Wording:  "Limit exceeded for the requested AWS resource"

File: AmplifyPlugins/Auth/AWSCognitoAuthPluginTests/AuthUserBehaviorTests/UserBehaviorChangePasswordTests.swift (hunk 2 above)
Comment (thisisabhash) [reply AmplifyPlugins/Auth/AWSCognitoAuthPluginTests/AuthUserBehaviorTests/UserBehaviorChangePasswordTests.swift:154]: There is a `testChangePasswordWithTooManyRequestsException` in the tests. Similarly, in other files changed there is a test for `requestLimitExceeded` available.

**Review by royjit — APPROVED** [review]

**Pull request merged**